### ISSUE TITLE
[Feature] Better logs in trace viewer for waitForRequest/waitForResponse

### ISSUE THREAD

**Issue by A-ZC-Lau** [microsoft/playwright#19282]
Is it possible to get the request and response from a `waitForRequest` and `waitForResponse` in the trace viewer? Because right now, it really lacks any information to help debug.

It doesn't even have information on what URL it fulfilled

<img width="301" alt="Screenshot 2022-12-06 at 11 44 27 am" src="https://user-images.githubusercontent.com/20273647/205778357-368fde6d-891a-407a-a0cb-ae6b6aae650f.png">

**Comment by aslushnikov** [comment]
@A-ZC-Lau what language are you using? Is it Java?

**Comment by A-ZC-Lau** [comment]
@aslushnikov javascript

**Comment by aslushnikov** [comment]
@A-ZC-Lau 

I tried the following simple test:

```ts
// a.spec.ts
import { test, expect } from '@playwright/test';

test('should work', async ({ page }) => {
  await Promise.all([
    page.goto('https://example.com'),
    page.waitForRequest('https://example.com'),
  ]);
});
```
And I ended up with the following trace that has a URL:

<img width="909" alt="Screenshot 2022-12-06 at 3 01 48 PM" src="https://user-images.githubusercontent.com/746130/206062052-cdaddbcc-d0e9-4bf4-8ab6-33e6a366d234.png">

Could you please try this? Does it work for you?

**Comment by A-ZC-Lau** [comment]
@aslushnikov I'm running my tests using the function way

```
import {test} from "@playwright/test";

const api = "https://www.airbnb.com.au/api/v2/user_markets";


test("waitForRequest", async function ({page}) {
	const r = new RegExp(api);
	await page.route(r, route => {
		route.fulfill({
			body : "",
			status : 200,
		});
	});

	const [req] = await Promise.all([
		page.waitForRequest(route => {
			const url = route.url().match(r);
			const method = route.method() === "GET";
			return !!url && method;
		}),
		page.goto("https://www.airbnb.com"),
	]);
});
```

But even with the string parameter, it doesn't show what the request/response (for waitForResponse) were.

Would it be possible to have these details added in future versions?

**Comment by aldenquimby** [comment]
I landed here from Google trying to find #15690

Commenting in case future travelers are in the same boat

**Comment by pavelfeldman** [comment]
#### Why was this issue closed?

Thank you for your involvement. This issue was closed due to limited engagement (upvotes/activity), lack of recent activity, and insufficient actionability. To maintain a manageable database, we prioritize issues based on these factors.

If you disagree with this closure, please open a new issue and reference this one. More support or clarity on its necessity may prompt a review. Your understanding and cooperation are appreciated.
